### PR TITLE
Store estimated models for nuisance parameters

### PR DESCRIPTION
**Pull request by MalteKurz**
### Description
This PR implements the often requested feature to store the estimated models for nuisance parameters. To use it, call the method `fit()` with option `store_models=True`. Example:
```R
library(DoubleML)
library(mlr3)
library(mlr3learners)
library(data.table)
set.seed(2)
ml_g = lrn("regr.ranger", num.trees = 10, max.depth = 2)
ml_m = ml_g$clone()
obj_dml_data = make_plr_CCDDHNR2018(alpha = 0.5)
dml_plr_obj = DoubleMLPLR$new(obj_dml_data, ml_g, ml_m)
dml_plr_obj$fit(store_models=TRUE)
```

The estimated models can then be found in the attribute `dml_plr_obj$models`:
```
dml_plr_obj$models
$ml_l
$ml_l$d
$ml_l$d[[1]]
$ml_l$d[[1]][[1]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_l$d[[1]][[2]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_l$d[[1]][[3]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_l$d[[1]][[4]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_l$d[[1]][[5]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights




$ml_m
$ml_m$d
$ml_m$d[[1]]
$ml_m$d[[1]][[1]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_m$d[[1]][[2]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_m$d[[1]][[3]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_m$d[[1]][[4]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights

$ml_m$d[[1]][[5]]
<LearnerRegrRanger:regr.ranger>
* Model: ranger
* Parameters: num.threads=1, num.trees=10, max.depth=2
* Packages: mlr3, mlr3learners, ranger
* Predict Types:  [response], se
* Feature Types: logical, integer, numeric, character, factor, ordered
* Properties: hotstart_backward, importance, oob_error, weights
```
Note that the number of fitted models depends on the settings and the considered model. The outer named list contains one entry for each nuisance part (here `ml_l` and `ml_m`). For each nuisance part there is a named list containing an entry for each treatment variable (here only `'d'`). The next inner part is a list of length `n_rep` (repeated cross-fitting) and then a list of length `n_folds` (number of folds per repeated cross fit).

### PR Checklist

- [x] The title of the pull request summarizes the changes made.
- [x] The PR contains a detailed description of all changes and additions.
- [x] The code passes `R CMD check` and all (unit) tests (see our [contributing guidelines](https://github.com/DoubleML/doubleml-for-r/blob/master/CONTRIBUTING.md#checklist-for-pull-requests-pr) for details).
- [x] Enhancements or new feature are equipped with unit tests.
- [x] The changes adhere to the "mlr-style" standards (see our [contributing guidelines](https://github.com/DoubleML/doubleml-for-r/blob/master/CONTRIBUTING.md#checklist-for-pull-requests-pr) for details).

### CODE REVIEW

File: R/double_ml_pliv.R [hunk 1]
```diff
@@ -510,19 +510,21 @@ DoubleMLPLIV = R6Class("DoubleMLPLIV",
                 est_params = self$get_params(paste0("ml_m_", x)),
                 return_train_preds = FALSE,
                 task_type = private$task_type$ml_m,
-                fold_specific_params = private$fold_specific_params)
+                fold_specific_params = private$fold_specific_params)$preds
             }))
+        # TODO: Export of fitted models not implemented for this case
```
Comment (PhilippBach) [inline R/double_ml_pliv.R:515]: I see. I guess it's alright to not support model export here. The IV models (with multiple Z) is also not yet final, so we should do it once we have a more mature version of them here

**Comment by japui-coder** [comment]
Store estimated models for nuisance parameters #169

**Review by japui-coder — CHANGES_REQUESTED** [review]
tests/testthat/test-double_ml_plr_export_preds.R

**Review by japui-coder — APPROVED** [review]
tests/testthat/test-double_ml_active_bindings.R

**Pull request merged**